compiler-rt/lib/radsan/radsan_context.cpp

```diff
@@ -0,0 +1,87 @@
+//===--- radsan_context.cpp - Realtime Sanitizer --------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+//===----------------------------------------------------------------------===//
+
+#include <radsan/radsan_context.h>
+
+#include <radsan/radsan_stack.h>
+
+#include <sanitizer_common/sanitizer_allocator_internal.h>
+#include <sanitizer_common/sanitizer_stacktrace.h>
+
+#include <new>
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+using namespace __sanitizer;
+
+namespace detail {
+
+static pthread_key_t Key;
+static pthread_once_t KeyOnce = PTHREAD_ONCE_INIT;
+void internalFree(void *Ptr) { __sanitizer::InternalFree(Ptr); }
```

Comment:
I wonder if this function definition is overkill here, and we could just inline `__sanitizer::InternalFree` at line 35 now

Reply:
I can test this out btw

Reply:
This is not optional, the InternalFree has a second arg which pthread_key_create doesn't respect:

void internalFree(void *Ptr) 

vs

void InternalFree(void *p, InternalAllocatorCache *cache = nullptr);

Reply:
Aha. Good spot. Should have trusted our earlier decision making! 😅